### ISSUE TITLE
model/lstm_attention.py

### ISSUE THREAD

**Issue by temur3559** [temur3559/UltimateCryptoBot#6]
import torch
import torch.nn as nn

class Attention(nn.Module):
    def __init__(self, hidden_dim):
        super(Attention, self).__init__()
        self.attn = nn.Linear(hidden_dim, 1)

    def forward(self, lstm_output):
        weights = torch.softmax(self.attn(lstm_output), dim=1)
        context = torch.sum(weights * lstm_output, dim=1)
        return context, weights

class LSTMAttentionModel(nn.Module):
    def __init__(self, input_dim, hidden_dim, output_dim, num_layers=1):
        super(LSTMAttentionModel, self).__init__()
        self.lstm = nn.LSTM(input_dim, hidden_dim, num_layers, batch_first=True)
        self.attention = Attention(hidden_dim)
        self.fc = nn.Linear(hidden_dim, output_dim)

    def forward(self, x):
        lstm_out, _ = self.lstm(x)
        context, _ = self.attention(lstm_out)
        out = self.fc(context)
        return out

# Test uchun (offline test qilishda ishlatiladi)
if __name__ == "__main__":
    model = LSTMAttentionModel(input_dim=5, hidden_dim=64, output_dim=1)
    dummy_input = torch.randn(32, 50, 5)  # 32 ta sample, har biri 50 candle, har birida 5 feature
    output = model(dummy_input)
    print("Model output shape:", output.shape)
